src/lang/modifyAst/addFillet.ts

```diff
@@ -0,0 +1,182 @@
+import {
+  ArrayExpression,
+  CallExpression,
+  ObjectExpression,
+  PathToNode,
+  PipeExpression,
+  Program,
+  VariableDeclaration,
+  VariableDeclarator,
+} from '../wasm'
+import {
+  createCallExpressionStdLib,
+  createLiteral,
+  createPipeSubstitution,
+  createObjectExpression,
+  createArrayExpression,
+  findUniqueName,
+} from '../modifyAst'
+import { getNodeFromPath } from '../queryAst'
+import { KCL_DEFAULT_CONSTANT_PREFIXES } from 'lib/constants'
+
+export function addFilletToEdgeWithinExtrude(
+  node: Program,
+  pathToSegmentNode: PathToNode,
+  pathToExtrudeNode: PathToNode,
+  filletRadius: number = 5
+): { modifiedAst: Program; pathToSegmentNode: PathToNode } {
+  // Clone AST
+  const _node = { ...node }
+
+  /**
+   * Add Tag to the Line Expression
+   */
+
+  // Find the specific line segment to tag with the new tag
+  const lineSegment = getNodeFromPath<CallExpression>(
+    _node,
+    pathToSegmentNode,
+    'CallExpression'
+  ).node
+
+  // Tag the line segment if it is not already tagged
+  function isLiteral(node: any): node is { type: 'Literal'; value: any } {
+    return node.type === 'Literal'
+  }
+
+  let tag = null
+  if (lineSegment.callee.name === 'line') {
+    if (lineSegment.arguments.length === 2) {
+      tag = findUniqueName(node, KCL_DEFAULT_CONSTANT_PREFIXES.SEGMENT, 2)
+      lineSegment.arguments.push(createLiteral(tag))
+    } else if (lineSegment.arguments.length === 3) {
+      if (isLiteral(lineSegment.arguments[2])) {
+        tag = lineSegment.arguments[2].value as string
+      } else {
+        throw new Error('Expected a Literal node')
+      }
+    } else {
+      throw new Error('Line segment not found.')
+    }
+  } else {
+    throw new Error('Line segment not found.')
+  }
```

Comment:
This makes a couple of assumtions, on that it's a `line` stdlib function call, and that the tag is in the 3rd positon (which ofc makes sense for line) but really we want this to be flexible for all of the stdlib segment functions. The good news is that all of the `sketchLineHelpers` in `src/lang/std/sketch.ts` have a util function `addTag` that knows how to add a tag for each stdlib segment function. (It's always the 3rd argument, with the exception of `close` which it's the second, but there could be more in future where it's different.)

All of these helper function can be combine by using the `sketchLineHelpersMap`,  you should be able to use `addTagForSketchOnFace` or at least take inspiration from it.

There was a typing tweak I wanted to make debf270333c39900819a95a0608fd4dfe7f1bd3f

Reply:
thanks, good point, I've swapped the whole chunk with a:

```ts
const { tag } = addTagForSketchOnFace(
    {
      pathToNode: pathToSegmentNode,
      node: node,
    },
    sketchSegment.callee.name
  )
```
and it works just fine :)

Reply:
Okay sweet, we should probably rename that function then haha 😝